### PR TITLE
[WebHost] Add support for items-list, locations-list, and custom-list option types to weighted-settings

### PR DESCRIPTION
**Pull request by LegendaryLinux**
## What is this fixing or adding?
This PR adds support for `item-list`, `location-list`, and `custom-list` option types to the Weighted Settings page.

## How was this tested?
Tested locally on my development machine.

## If this makes graphical changes, please attach screenshots.
All three list types look the same, but here's a screenshot of each type anyway.

Locations List (Timespinner):
image
image

Items List (Kingdom Hearts 2):
image
image

Custom List (Timespinner):
image
image

### CODE REVIEW

**Comment by Berserker66** [comment]
Looking at the listing, the handling for Priority Locations and Exclude Locations seems a bit confused.

Priority Locations are listed on their own early on and then Exclude Locations are part of the end special handling.
The two should probably be next to each other. Also a location can't be both priority and excluded, while that won't crash, it might allow for better UI to express it there.

**Comment by LegendaryLinux** [comment]
Are `proirity_locations` and `exclude_locations` present on every game? Could they be hardcoded similar to how `random`, `random-low` and `random-high` are for other options?

**Comment by LegendaryLinux** [comment]
Fixed the text in the priority & exclusion items headers, and swapped the two positions.
image

File: WebHostLib/static/assets/weighted-settings.js [hunk 1]
```diff
@@ -791,29 +894,89 @@ const buildHintsDiv = (game, items, locations) => {
...
+
+  const locationsDiv = document.createElement('div');
+  locationsDiv.classList.add('locations-div');
+  const locationsHeader = document.createElement('h3');
+  locationsHeader.innerText = 'Priority & Exclusion Locations';
+  locationsDiv.appendChild(locationsHeader);
+  const locationsDescription = document.createElement('p');
+  locationsDescription.classList.add('setting-description');
+  locationsDescription.innerText = 'Priority locations guarantee a progression item will be placed there while ' +
+    'excluded locations will not contain progression items.';
```
Comment (KonoTyran) [inline WebHostLib/static/assets/weighted-settings.js:927]: Proabby want to add that is also stops `useful` items from being in these locations.

Comment (LegendaryLinux) [reply WebHostLib/static/assets/weighted-settings.js:927]: Okay, I'll do that in a bit here.

**Review by KonoTyran — APPROVED** [review]

**Pull request merged**